ykcompile/src/lib.rs

```diff
@@ -106,6 +106,7 @@ enum Location {
     Register(u8),
     Stack(i32),
     Arg(i32),
+    Deref(Box<Location>),
```

Comment:
Perhaps not for this PR, but IIRC we agreed that all of the projections would fall under one "Projection" variant?

I'm also still wondering if `Stack(offset)` should be `Mem(reg, offset)` to handle things like heap allocs or offsets from temporary registers, but again, possibly for another PR.

Reply:
I agree. One thing about `Deref`s though is that I think it needs to be recursive, which is why I just made it part of `Location`.

I always assumed that some of these things will change later on. It's hard to say what this stuff needs to look like until we have actual examples (e.g. heap alloc) to refer to. And we can't really solve everything in one go. So we'll have to come back to it in another PR.

Reply:
Yep, that's fine.

Let's be on the lookout for variants in Location that can be generalised later.